Adds openmp offload for XSBench, RSBench, miniFE

koparasy opened the pull request:
This patch adds compiling configurations for XSBench, RSBench, and miniFE using the OpenMP offload programming model.

koparasy committed:
Adds openmp offload for XSBench, RSBench, miniFE

ggeorgakoudis commented on config.yaml:
```diff
@@ -106,20 +106,24 @@ miniAMR:
     run: 'env OMP_NUM_THREADS=1 OMP_PROC_BIND=true ./XSBench'
-    input: '-t 1 -k 1 -s small'
-    measure:
+    input: '-t 1 -m event -s small'
```
Is this input compatible with the other configurations (omp, seq)?

ggeorgakoudis commented on config.yaml:
```diff
@@ -135,12 +139,15 @@ RSBench:
     run: 'env OMP_NUM_THREADS=1 OMP_PROC_BIND=true ./rsbench'
-    input: '-t 1 -s small'
-    measure:
+    input: '-s small -m event -t 1'
```
Is this input compatible with the other configurations (seq, omp)?

ggeorgakoudis commented on config.yaml:
```diff
@@ -183,6 +190,14 @@ miniFE:
         ],
+        omp-offload: [ 'cd openmp45/src; 
+        make -j CC=clang CXX=clang++ 
+        CFLAGS="-v -O3 -fopenmp -fopenmp-targets=nvptx64-nvidia-cuda 
```
Remove -v

ggeorgakoudis commented on patches/RSBench.patch:
```diff
@@ -1,3 +1,36 @@
+ 				print_CLI_error();
+ 		}
+-		else
++        else if( strcmp(arg, "-t") == 0 )
```
Is this to enable multi-threaded target creation? Is an option on the number of threads correct? Shouldn't be 1 for correct program execution?

ggeorgakoudis commented on patches/XSBench.patch:
```diff
@@ -0,0 +1,22 @@
++		}
+ 		// n_gridpoints (-g)
+-		if( strcmp(arg, "-g") == 0 )
++        else if( strcmp(arg, "-g") == 0 )
```
Change to if, otherwise -t precludes -g

ggeorgakoudis commented on patches/XSBench.patch:
```diff
@@ -0,0 +1,22 @@
+ 		char * arg = argv[i];
+ 
++		// This will be ignored
++		if( strcmp(arg, "-t") == 0 )
```
Is an option on the number of threads correct? Shouldn't be 1 for correct program execution?

ggeorgakoudis commented on config.yaml:
```diff
@@ -106,20 +106,24 @@ miniAMR:
 
 XSBench:
-    fetch: 'git clone https://github.com/ANL-CESAR/XSBench.git'
+    fetch: 'git clone https://github.com/ANL-CESAR/XSBench.git;
```
Please add the command to clone/checkout the specific commit that this patch works with. Also for RSBench fetch. We should do it for every repo that needs a patch.

ggeorgakoudis reviewed (changes requested):
Please reply to comments and address issues

ggeorgakoudis commented:
Possibly add:

miniQMC
QMCpack
gridmini
SPEC accel(?)
SPEC HPG (?)

koparasy committed:
OpenMP offload support.

- Move offload applications to another configuration file.
- Add Clang and IBM compilation toolchain for applications.
- Extends Harness to parse information out of the nvlink info
- The parsed data are stored under the results/results-'application name'.yaml file
- Add miniQMC application

koparasy committed:
Creates results directory before building application
